### ISSUE TITLE
Object properties that use ES6 shorthand are renamed.

### ISSUE THREAD

**Issue by Permutatrix** [rollup/rollup#575]
If Rollup renames a variable that's used in an object literal as a shorthand property, the name of the property will be changed, too.

For instance, this:
```js
const something = 5;
return { something };
```
May be changed to this:
```js
const something$$ = 5;
return { something$$ };
```
I would expect it to become this instead:
```js
const something$$ = 5;
return { something: something$$ };
```

This behavior breaks the API of a module I'm working on. The transpiled version of [this](https://github.com/Permutatrix/whether-manager/blob/d7b20a53c93557db76c479b08f73c8db5761f988/src/node.js#L33-L35) after being run through Rollup and Babel looks like this:
```js
self = {
  name: name, set: set, remove$$: remove$$, has: has, get: get, nodes: nodes, getNodes: getNodes
};
```
As you can see, Rollup has changed the name of the `remove` method to `remove$$`.

**Comment by Rich-Harris** [comment]
Are you able to reproduce this on rollupjs.org? I can't seem to –  the shorthand is being expanded properly ([demo](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22es6%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22import%20bar%20from%20'.%2Fbar.js'%3B%5Cnimport%20foo%20from%20'.%2Ffoo.js'%3B%5Cn%5Cnconsole.log(%20foo().something%20)%3B%5Cnconsole.log(%20bar().something%20)%3B%22%7D%2C%7B%22name%22%3A%22foo.js%22%2C%22code%22%3A%22const%20something%20%3D%20'foo'%3B%5Cnexport%20default%20function%20()%20%7B%5Cn%20%20return%20%7B%20something%20%7D%3B%5Cn%7D%22%7D%2C%7B%22name%22%3A%22bar.js%22%2C%22code%22%3A%22const%20something%20%3D%20'bar'%3B%5Cnexport%20default%20function%20()%20%7B%5Cn%20%20return%20%7B%20something%20%7D%3B%5Cn%7D%22%7D%5D%7D))

**Comment by Permutatrix** [comment]
@Rich-Harris Yes, I copied the full code of whether-manager in and got the same incorrect result.

In your demo, the variable is global and is renamed by appending `$1`, whereas in mine, it's local and renamed by appending `$$`. Perhaps those cases are handled differently?

**Comment by Rich-Harris** [comment]
When I say 'are you able to reproduce it' I mean 'please reproduce it' :grinning: We can't fix the bug unless we can see it, and googling 'whether-manager' doesn't turn up anything useful

**Comment by Permutatrix** [comment]
@Rich-Harris Yeah, sorry. I was working on it. [Here you go.](http://rollupjs.org/#{%22options%22%3A{%22format%22%3A%22es6%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A{}}%2C%22modules%22%3A[{%22name%22%3A%22main.js%22%2C%22code%22%3A%22import%20foo%20from%20%27.%2Ffoo.js%27%3B\n\nexport%20default%20function%20bar%28%29%20{\n%20%20const%20something%20%3D%20foo%28%29%3B\n%20%20return%20{%20something%20}%3B\n}\n%22}%2C{%22name%22%3A%22foo.js%22%2C%22code%22%3A%22export%20default%20function%20something%28%29%20{\n%20%20\n}%22}]})

[whether-manager](https://github.com/Permutatrix/whether-manager) is the module I'm working on that's affected by this bug. I linked to it briefly in the original report and forgot that people don't usually have an encyclopedic knowledge of everything they read.

**Comment by Permutatrix** [comment]
@Rich-Harris [Okay, I fixed the link.](http://rollupjs.org/#{%22options%22%3A{%22format%22%3A%22es6%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A{}}%2C%22modules%22%3A[{%22name%22%3A%22main.js%22%2C%22code%22%3A%22import%20foo%20from%20%27.%2Ffoo.js%27%3B%5Cn%5Cnexport%20default%20function%20bar%28%29%20{%5Cn%20%20const%20something%20%3D%20foo%28%29%3B%5Cn%20%20return%20{%20something%20}%3B%5Cn}%5Cn%22}%2C{%22name%22%3A%22foo.js%22%2C%22code%22%3A%22export%20default%20function%20something%28%29%20{%5Cn%20%20%5Cn}%22}]})

**Comment by Rich-Harris** [comment]
Ah, I see. Thanks

**Comment by Rich-Harris** [comment]
Fixed in 0.25.7
